src/lib/draw/EditMode.svelte

```diff
@@ -41,28 +41,23 @@
     unsavedFeature = JSON.parse(JSON.stringify(feature));
 
     if (feature.geometry.type == "LineString") {
-      // TODO Update route-snapper-ts to use Except<route_name> or otherwise pick the important properties
-      $routeTool?.editExistingRoute(
-        feature as unknown as Feature<LineString, RouteProps>,
-      );
-      $routeTool?.addEventListenerSuccess(onSuccess);
-      $routeTool?.addEventListenerUpdated(onUpdate);
-      $routeTool?.addEventListenerFailure(onFailure);
+      // Transform into the correct format
+      $routeWaypoints = feature.properties.waypoints!.map((waypt) => {
+        return {
+          point: [waypt.lon, waypt.lat],
+          snapped: waypt.snapped,
+        };
+      });
       controls = "route";
     } else if (feature.geometry.type == "Polygon") {
-      if (feature.properties.waypoints) {
-        $routeTool?.editExistingArea(feature as Feature<Polygon, AreaProps>);
-        $routeTool?.addEventListenerSuccess(onSuccess);
```

Comment:
Do we want to now delete these functions? looks like they're no longer used to me

Reply:
`addEventListenerSuccess` and such? They're in the upstream package https://github.com/dabreegster/route_snapper/blob/main/route-snapper-ts/src/index.ts. We essentially stop using `route-snapper-ts` in this PR, using part of the WASM API directly. I'm still depending on route-snapper-ts here for now to get `RouteProps` and a few convenient definitions, but could maybe clean that up later too